packages/aws-cdk-lib/aws-stepfunctions-tasks/lib/bedrock/create-model-customization-job.ts

```diff
@@ -0,0 +1,396 @@
+import { Aws, Stack, Token } from '../../../core';
+import { integrationResourceArn, validatePatternSupported } from '../private/task-utils';
+
+/**
+ * The customization type.
+ *
+ * @see https://docs.aws.amazon.com/bedrock/latest/userguide/custom-models.html
+ */
+export enum CustomizationType {
+  /**
+   * Fine-tuning
+   *
+   * Provide labeled data in order to train a model to improve performance on specific tasks.
+   */
+  FINE_TUNING = 'FINE_TUNING',
+  /**
+   * Continued pre-training
+   *
+   * Provide unlabeled data to pre-train a foundation model by familiarizing it with certain types of inputs.
+   */
+  CONTINUED_PRE_TRAINING = 'CONTINUED_PRE_TRAINING',
+}
+
+/**
+ * The key/value pair for a tag.
+ */
+export interface ITag {
+  /**
+   * Key for the tag.
+   */
+  readonly key: string;
+  /**
+   * Value for the tag.
+   */
+  readonly value: string;
+}
+
+/**
+ * VPC configuration
+ */
+export interface IBedrockCreateModelCustomizationJobVpcConfig {
+  /**
+   * VPC configuration security groups
+   */
+  readonly securityGroups: ec2.ISecurityGroup[];
+  /**
+   * VPC configuration subnets
+   */
+  readonly subnets: ec2.ISubnet[];
+}
+
+/**
+ * Properties for invoking a Bedrock Model
+ */
+export interface BedrockCreateModelCustomizationJobProps extends sfn.TaskStateBaseProps {
+  /**
+   * The base model.
+   */
+  readonly baseModel: bedrock.IModel;
+  /**
+   * A unique, case-sensitive identifier to ensure that the API request completes no more than one time.
+   * If this token matches a previous request, Amazon Bedrock ignores the request, but does not return an error.
+   *
+   * @see https://docs.aws.amazon.com/AWSEC2/latest/APIReference/Run_Instance_Idempotency.html
+   *
+   * @default - no client request token
+   */
+  readonly clientRequestToken?: string;
+  /**
+   * The customization type.
+   *
+   * @default FINE_TUNING
+   */
+  readonly customizationType?: CustomizationType;
+  /**
+   * The custom model is encrypted at rest using this key.
+   *
+   * @default - no encryption
+   */
+  readonly kmsKey?: kms.IKey;
+  /**
+   * A name for the resulting custom model.
+   */
+  readonly customModelName: string;
+  /**
+   * Tags to attach to the resulting custom model.
+   *
+   * @default - no tags
+   */
+  readonly customModelTags?: ITag[];
+  /**
+   * Parameters related to tuning the model.
+   *
+   * @see https://docs.aws.amazon.com/bedrock/latest/userguide/custom-models-hp.html
+   *
+   * @default - use default hyperparameters
+   */
+  readonly hyperParameters?: { [key: string]: string };
+  /**
+   * A name for the fine-tuning job.
+   */
+  readonly jobName: string;
+  /**
+   * Tags to attach to the job.
+   *
+   * @default - no tags
+   */
+  readonly jobTags?: ITag[];
+  /**
+   * The S3 URI where the output data is stored.
+   *
+   * @see https://docs.aws.amazon.com/bedrock/latest/APIReference/API_OutputDataConfig.html
+   */
+  readonly outputDataS3Uri: string;
```

Comment:
```suggestion
  readonly outputDataS3: s3.IBucket;
```
What about passing a bucket and using the [`s3UrlForObject`](https://github.com/aws/aws-cdk/blob/df48fd781a476ada79b6b49f006a60f33f73d409/packages/aws-cdk-lib/aws-s3/lib/bucket.ts#L169) method to retrieve the S3 URI?
Same for other `...S3Uri` props. 
We can then remove the `s3UriToArn` private method.

Reply:
It sounds nice! 

My concern is it may be difficult to specify S3 path.
Ex. To specify s3://someBucket/train as training data and s3://someBucket/validate as validation data.

What is your thought about this? I think it is acceptable to receive these props as IBucket.

Reply:
Good point!
However, I'm not a fan of the `s3UriToArn` method, I find it a bit too complicated.
Maybe adding an `outputDataS3Prefix?: string` property to specify the S3 path could be a solution (same for the other 2 buckets).
What's your take?

Reply:
I agree with your suggestion and will try to modify the code to accept `IBucket` and `prefix` as arguments respectively.

I have a question for consultation. If `validationDataS3` is received as an array of `IBucket`, should I also receive prefix as an array of strings and associate elements of the same index?

Ex. 
```ts
{
  ...,
  validationDataS3: [bucket1, bucket2],
  validationDataPrefix: ['prefix1', 'prefix2']
}
```

Reply:
🤔 Yeah, that would be annoying to manage.
What about adding an `interface` for the bucket + prefix pair ([example](https://github.com/aws/aws-cdk/blob/bf2cf51bf1e8458097b23e7f580ad0978b7e4cca/packages/aws-cdk-lib/aws-synthetics/lib/canary.ts#L80))?

Reply:
Thank you for your nice suggestion.

I've defined `BucketConfiguration` and updated training, validation and output data properties for receiving `BucketConfiguration`.